docs/client-side-architecture.md

```diff
@@ -0,0 +1,128 @@
+Cozy Konnectors client-side app architecture
+============================
+
+Current state
+-------------
+
+- CoffeeScript codebase
+- Backbone app
+- Simple structure
+- Oignon architecture
+- Hard-coded templates structures (low flexibility)
+- Nested styles
+- Bugged realtime
+- Not a core to power konnectors, but not a full opionated plateform
+
+
+Should we keep it?
+------------------
+
+### Pros
+
+- it exists
+- it is used for dozen of konnectors
+- it doesn't need particular skills about the plateform to add a Konnector
+- it uses a well-known architecture (just a Backbone with some custom base views)
+
+
+### Cons
+
+- unmaintainable in long-term (markups in views, too deeply nested concerns…)
+- need to revamp all styles and a huge part of markups
+- doesn't offer enough flexibility to build views
+- adding features (like OAuth) will be difficult due to hard-coded behaviors
+- realtime isn't sufficiently sustainable (takes place at root instead of being dedicated to components it relies on)
+
+### Recommendation
+
+As the app isn't really complicated, and because we wanted to reconsider the way we display / manage / add features to konnectors, we recommand to trash the actual version of the client-side app, and start a fresh new one. The actual code is too deeply messy to be really sustainable in future, and doesn't let community easily add features.
+
+
+Architecture
+------------
+
+### Overview
+
+The konnector client-side app should provides the following features:
+
+- a dedicated core to power konnectors as autonomous entities
+- a set of useful components to build konnectors views (username, password, directory-select, tokens…)
+- a default view for basic konnectors (username / password / directory-target)
+- keep each konnector separated from the core
+- a minimalist API to interact with core features (e.g. to submit / validate form)
+- an easy way to override default behaviors (logics, views, styles…) for a given konnector
+
+
+### Technical recommendation
+
+We suggest to build the new version of Konnector client-side app in ES6, using _Vue.js_ with _vue-router_ and _vue-resource_ for the app logics, and Webpack for the packaging tool along the Vue-loader webpack's loader.
```

Comment:
Btw, Phyks was asking on IRC if that made sense to develop konnectors using Vue.js whereas other cozy projects (email at least IIRC) are using React ?